modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/CompilerWrapper.java

```diff
@@ -117,6 +117,17 @@ public boolean isOutDated() {
 		return super.isOutDated();
 	}
 
+	@Override
```

Comment:
@dantewang @yuhai 
I think we want to keep the java files by default, see portal.properties for "sp.servlet.init.param.keepgenerated=true", but I am not sure why we set "defaults.put("keepgenerated", "false");" in JspServlet.

Also confusing why we have to remove it here.

Tina.

Reply:
@tinatian Here I override "removeGeneratedFiles" because the method of Tomcat Jasper blindly creates the scratch dir even if it doesn't need to remove anything.